sympy/matrices/common.py

```diff
@@ -203,12 +203,11 @@ def col_insert(self, pos, other):
         if not self:
             return type(self)(other)
 
+        i = pos
         if pos < 0:
             pos = self.cols + pos
-        if pos < 0:
-            pos = 0
-        elif pos > self.cols:
-            pos = self.cols
```

Comment:
I think these lines should be left as-is (and exception should not be raised).
This behaviour is consistent with the way indexing works in Python. 
The only change could be `i = as_int(pos)` (with the necessary import).

Reply:
@sidhantnagpal 
>The only change could be i = as_int(pos) (with the necessary import).

I will do that.
>I think these lines should be left as-is (and exception should not be raised).

Why do you think a float input should not result in an error?

Reply:
Those lines don't handle float inputs specially.
The conditional checks could use `i` instead of `pos`.
The exception will then be raised in `as_int` (and not anywhere else).